hp_list/main.c

```diff
@@ -14,6 +14,110 @@
 #include <string.h>
 #include <threads.h>
 
+#ifdef RUNTIME_STAT
+/*
+ * "rtry" the number of retries in the __list_find function.
+ * "cons" the number of wait-free contains in the __list_find function that curr
+ * pointer pointed.
+ * "trav" the number of list element traversal in the __list_find function.
+ * "fail" the number of CAS() failures.
+ * "del" the number of list_delete operation failed and restart again.
+ * "ins" the number of list_insert operation failed and restart again.
+ */
+static atomic_uint_fast64_t rtry = 0, cons = 0, trav = 0, fail = 0;
+static atomic_uint_fast64_t del = 0, ins = 0;
+static atomic_uint_fast64_t deletes = 0, inserts = 0;
+
+#define TRACE_GOTO_TRY_AGAIN        \
```

Comment:
The macro `TRACE_GOTO_TRY_AGAIN` should not perform `goto try_again;`.

Instead, we can use the following sequence:
```c
TRACE_ATOMIC_TRIES();
goto try_again;
```